pkg/dynamicaccess/accesslogic_test.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2024 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package dynamicaccess_test
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"encoding/hex"
+	"testing"
+
+	"github.com/ethersphere/bee/v2/pkg/crypto"
+	"github.com/ethersphere/bee/v2/pkg/dynamicaccess"
+	kvsmock "github.com/ethersphere/bee/v2/pkg/kvs/mock"
+	"github.com/ethersphere/bee/v2/pkg/swarm"
+	"github.com/stretchr/testify/assert"
+)
+
+// Generates a new test environment with a fix private key
+func setupAccessLogic() dynamicaccess.ActLogic {
+	privateKey := getPrivKey(1)
+	diffieHellman := dynamicaccess.NewDefaultSession(privateKey)
+	al := dynamicaccess.NewLogic(diffieHellman)
+
+	return al
+}
+
+func getPrivKey(keyNumber int) *ecdsa.PrivateKey {
+	var keyHex string
+
+	switch keyNumber {
+	case 0:
+		keyHex = "a786dd84b61485de12146fd9c4c02d87e8fd95f0542765cb7fc3d2e428c0bcfa"
+	case 1:
+		keyHex = "b786dd84b61485de12146fd9c4c02d87e8fd95f0542765cb7fc3d2e428c0bcfb"
+	case 2:
+		keyHex = "c786dd84b61485de12146fd9c4c02d87e8fd95f0542765cb7fc3d2e428c0bcfc"
+	default:
+		panic("Invalid key number")
+	}
+
+	data, err := hex.DecodeString(keyHex)
+	if err != nil {
+		panic(err)
+	}
+
+	privKey, err := crypto.DecodeSecp256k1PrivateKey(data)
+	if err != nil {
+		panic(err)
+	}
+
+	return privKey
+}
+
+func TestDecryptRef_Success(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	id1 := getPrivKey(1)
+	s := kvsmock.New()
+	al := setupAccessLogic()
+	err := al.AddGrantee(ctx, s, &id1.PublicKey, &id1.PublicKey)
+	if err != nil {
+		t.Fatalf("AddGrantee: expected no error, got %v", err)
+	}
+
+	byteRef, _ := hex.DecodeString("39a5ea87b141fe44aa609c3327ecd896c0e2122897f5f4bbacf74db1033c5559")
+	expectedRef := swarm.NewAddress(byteRef)
+	encryptedRef, err := al.EncryptRef(ctx, s, &id1.PublicKey, expectedRef)
+	if err != nil {
+		t.Fatalf("There was an error while calling EncryptRef: %v", err)
+	}
+
+	actualRef, err := al.DecryptRef(ctx, s, encryptedRef, &id1.PublicKey)
+	if err != nil {
+		t.Fatalf("There was an error while calling Get: %v", err)
+	}
+
+	if !expectedRef.Equal(actualRef) {
+		t.Fatalf("DecryptRef gave back wrong Swarm reference! Expedted: %v, actual: %v", expectedRef, actualRef)
+	}
+}
+
+func TestDecryptRefWithGrantee_Success(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	id0, _ := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
```

Comment:
Check for errors in all places. There are more than 50 places where error is not checked, use a helper function for convenience, but errors should be checked in tests.

Reply:
asserted all the errors in the tests